### PR TITLE
Adding support for check and diff mode with refactoring for Host module

### PR DESCRIPTION
**Pull request by trisha-dell**
# Description
Adding support for check and diff mode with refactoring for Host module

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| |

# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, pep8, linting, or security issues
- [x] I have performed Ansible Sanity test using --docker default
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate
- [x] I have maintained at least 90% code coverage
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Backward compatibility is not broken

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- [x] Unit test
- [x] Functional Test

### CODE REVIEW

File: plugins/modules/host.py [hunk 1]
```diff
@@ -755,69 +721,114 @@ def perform_module_operation(self):
...
-            LOG.info('Creating host %s', host_name)
-            changed = self.create_host(host_name)
-            if changed:
-                host_id = self.get_host_id_by_name(host_name)
+    def validate_create(self, initiator_state):
+        if self.module.params['new_name']:
+            error_msg = "Operation on host failed as new_name is given " \
+                        "for a host that doesnt exist."
+            LOG.error(error_msg)
+            self.module.fail_json(msg=error_msg)
```
Comment (Kritika-Bhateja-03) [inline plugins/modules/host.py:731]: These lines can be taken outside if as it is repeated

File: plugins/modules/host.py [hunk 2]
```diff
@@ -746,6 +711,7 @@ def perform_module_operation(self):
                         "provide a valid value."
             LOG.error(error_msg)
             self.module.fail_json(msg=error_msg)
```
Comment (Kritika-Bhateja-03) [inline plugins/modules/host.py:713]: We can shuffle the if conditions and put this common code at end so that the code is not duplicated

**Review by Kritika-Bhateja-03 — CHANGES_REQUESTED** [review]

**Review by ABHISHEK-SINHA10 — DISMISSED** [review]

**Review by sachin-apa — APPROVED** [review]

File: plugins/modules/host.py [hunk 3]
```diff
@@ -531,101 +540,83 @@ def _get_remove_initiators(self, existing, requested):
...
-                return True
+    def _prepare_add_list_with_type(self, add_list, detailed_initiators):
+        add_list_with_type = []
+        for init in add_list:
+            # when detailed_initiators param is used to add new initiators
+            if detailed_initiators:
+                for detailed_init in detailed_initiators:
+                    if init == detailed_init['port_name']:
+                        current_initiator = {}
+                        current_initiator['port_name'] = init
```
[the remark this answers is not in the document]
Comment (trisha-dell) [reply plugins/modules/host.py:563]: can't be called in as in if it is part of loop and is getting initiated for every instance.

**Review by rajshekarp87 — APPROVED** [review]

File: plugins/modules/host.py (hunk 2 above)
Comment (trisha-dell) [reply plugins/modules/host.py:713]: Not applicable as the code may not go in either scenario, hence no failure.

**Review by sachin-apa — APPROVED** [review]

**Review by rajshekarp87 — APPROVED** [review]

**Comment by sachin-apa** [comment]
@Kritika-Bhateja-03 i'm dismissing your review as from the comments they doesn't look like mandatory changes to be done or can be done and we need to merge the PR.

**Review by ABHISHEK-SINHA10 — APPROVED** [review]

**Pull request merged**